### PR TITLE
chore: bumpenv update NGC base images version to 24.03

### PR DESCRIPTION
**Pull request by jgongd**
## Description

Update the base image of `pytorch-ngc`, `pytorch-ngc-hpc`, `tensorflow-ngc`, and `tensorflow-ngc-hpc` to 24.03 to address critical vulnerabilities.

Environments repo PR: https://github.com/determined-ai/environments/pull/253/files

## Test Plan

CI tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by MikhailKardash — APPROVED** [review]
